src/GPG.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PharIo\GnuPG;
+
+interface GPG
+{
+    /**
+     * @param string $keyData raw key data
+     *
+     * @throws InvalidHomeDirectory
+     * @throws InvalidKey
+     */
+    public function importPublicKey(string $keyData): PublicKey;
```

Comment:
Yes, you are absolutely right... I should have looked better at the output of gpg here.

I think we should return the keyid's that are imported. So the return type would be 

```suggestion
    /** @return KeyId[] **/
    public function importPublicKey(string $keyData): array;
```

An import could contain multiple keyId when the imported file contains multiple keys. There is no difference between import of a private or publickey from a user point of view. But I think when we want to support passphrase protected keys we might want to make a difference.  Because the handling would be a bit different. the pecl library doesn't support this so people would need to use the cli wrapper.

Reply:
> Yes, you are absolutely right... I should have looked better at the output of gpg here.
> 
> I think we should return the keyid's that are imported. So the return type would be

Sorry to disappoint again: If we rely on the pecl extension, it appears that we only get the *count* of UIDs, not the list of them. Why every someone consider that useful to hide ;)

I'm more and more wondering whether having `ext/gnupg` support is actually worth it ;) But we should at least design the objects so they can work in both...

> 
> An import could contain multiple keyId when the imported file contains multiple keys. There is no difference between import of a private or publickey from a user point of view. But I think when we want to support passphrase protected keys we might want to make a difference. Because the handling would be a bit different. the pecl library doesn't support this so people would need to use the cli wrapper.

Fine with me.

Reply:
Regarding:

```suggestion
    /** @return KeyId[] **/
    public function importPublicKey(string $keyData): array;
```

Not happy yet with the signature either ;)
- I'm not a fan of `array` as a return type because it allows for all kinds of ugly edge cases
- We have more data (fingerprints for instance) that we'd possibly like to return
- With the pecl extension not providing a list of UIDs, this is probably not working
- What happend to the replacement of `string` with `PublicKeyData`? ;-)